### PR TITLE
Richli/check jsonl

### PR DESCRIPTION
**Pull request by RichardLi1437**
Add file match with wildcard support, to check if jsonl files available under specific folder structure.

### CODE REVIEW

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py [hunk 1]
```diff
@@ -67,65 +67,56 @@ def same_year(start, end) -> List[str]:
...
-                if store_url.is_folder_exists(f"{start.strftime('%Y')}") else []
+            return _get_files(f"{start.strftime('%Y')}/*/*/*/*.jsonl")
         return cross_month(start, end)
 
     def same_month(start, end) -> List[str]:
         if is_same_day(start, end):
             return same_day(start, end)
         if is_start_of_month(start) and is_end_of_month(end):
-            return [get_url(f"{start.strftime('%Y/%m')}/*/*/*.jsonl")] \
-                if store_url.is_folder_exists(f"{start.strftime('%Y/%m')}") else []
```
Comment (ycheng35xo) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py:79]: is_folder_exists still be used? If not, should we remove it?

File: assets/model_monitoring/components/tests/unit/test_store_url.py [hunk 2]
```diff
@@ -304,6 +305,87 @@ def test_is_local_path(self, path, is_local_path):
...
+    PATHS = [
+        "my/folder/folder1/folder2/file1.jsonl",
+        "my/folder/folder1/file2.jsonl",
+        "my/folder/folder3/file3.csv",
+        "my/folder/file4.jsonl",
+        "my/folder/folder1/folder2",
+        "my/folder/folder1"
+    ]
+
+    @pytest.mark.parametrize(
```
Comment (ycheng35xo) [inline assets/model_monitoring/components/tests/unit/test_store_url.py:317]: Is there a test case where the folder exists but contains no file?

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py (hunk 1 above)
Comment (RichardLi1437) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py:79]: not used by this scenario, but it's a useful StoreUrl utils for future use, so I choose to keep it there.

File: assets/model_monitoring/components/tests/unit/test_store_url.py (hunk 2 above)
Comment (RichardLi1437) [reply assets/model_monitoring/components/tests/unit/test_store_url.py:317]: added

**Review by ycheng35xo — DISMISSED** [review]

**Review by ycheng35xo — APPROVED** [review]

**Pull request merged**